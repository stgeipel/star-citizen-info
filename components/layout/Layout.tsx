import * as React from 'react';
import clsx from 'clsx';
import Head from 'next/head';
import AppHeader from './AppHeader';

export interface Props {
  title?: string;
  className?: string | Record<string, unknown>;
}

const Layout: React.FC<Props> = ({
  title = 'NextJS with TypeScript and TailwindCSS',
  className,
  children,
}) => {
  return (
    <>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="min-h-screen flex flex-col">
        <AppHeader />
        <main className={clsx('flex-grow', className)}>{children}</main>
        <footer className="text-xs text-center text-gray-600 py-4 px-3">
          Made with â™¥ by{' '}
          <a href="https://www.github.com/stgeipel">Steffen Geipel</a>
        </footer>
      </div>
    </>
  );
};

export default Layout;
