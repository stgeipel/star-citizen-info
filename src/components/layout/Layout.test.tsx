import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Layout from './Layout';

describe('Layout', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Layout>
        <h1>hello world</h1>
      </Layout>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot with title', () => {
    const wrapper = shallow(
      <Layout title="My Awesome App">
        <h1>hello world</h1>
      </Layout>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const wrapper = shallow(
      <Layout className="py-4 mx-3">
        <h1>hello world</h1>
      </Layout>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
