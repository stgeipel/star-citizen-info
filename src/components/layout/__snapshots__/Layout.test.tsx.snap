// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<Fragment>
  <Head>
    <title>
      NextJS with TypeScript and TailwindCSS
    </title>
  </Head>
  <div
    className="min-h-screen flex flex-col"
  >
    <AppHeader />
    <main
      className="flex-grow"
    >
      <h1>
        hello world
      </h1>
    </main>
    <footer
      className="text-xs text-center text-gray-600 py-4 px-3"
    >
      Made with ♥ by
       
      <a
        href="https://www.github.com/stgeipel"
      >
        Steffen Geipel
      </a>
    </footer>
  </div>
</Fragment>
`;

exports[`Layout matches snapshot with custom className 1`] = `
<Fragment>
  <Head>
    <title>
      NextJS with TypeScript and TailwindCSS
    </title>
  </Head>
  <div
    className="min-h-screen flex flex-col"
  >
    <AppHeader />
    <main
      className="flex-grow py-4 mx-3"
    >
      <h1>
        hello world
      </h1>
    </main>
    <footer
      className="text-xs text-center text-gray-600 py-4 px-3"
    >
      Made with ♥ by
       
      <a
        href="https://www.github.com/stgeipel"
      >
        Steffen Geipel
      </a>
    </footer>
  </div>
</Fragment>
`;

exports[`Layout matches snapshot with title 1`] = `
<Fragment>
  <Head>
    <title>
      My Awesome App
    </title>
  </Head>
  <div
    className="min-h-screen flex flex-col"
  >
    <AppHeader />
    <main
      className="flex-grow"
    >
      <h1>
        hello world
      </h1>
    </main>
    <footer
      className="text-xs text-center text-gray-600 py-4 px-3"
    >
      Made with ♥ by
       
      <a
        href="https://www.github.com/stgeipel"
      >
        Steffen Geipel
      </a>
    </footer>
  </div>
</Fragment>
`;
